tests/integration/ha/test_ha.py

```diff
@@ -0,0 +1,491 @@
+    )
+    await helpers.wait_idle(ops_test)
+
+
+@pytest.mark.abort_on_fail
+async def test_pod_reschedule(ops_test: OpsTest, request):
+    """Forcefully reschedules ZooKeeper pod."""
+    hosts = helpers.get_hosts(ops_test)
+    leader_name = helpers.get_leader_name(ops_test, hosts)
+    leader_host = helpers.get_unit_host(ops_test, leader_name)
+    password = helpers.get_super_password(ops_test)
+    parent = request.node.name
+    non_leader_hosts = ",".join([host for host in hosts.split(",") if host != leader_host])
+
+    logger.info("Starting continuous_writes...")
+    cw.start_continuous_writes(
+        parent=parent, hosts=hosts, username=helpers.USERNAME, password=password
+    )
+    await asyncio.sleep(CLIENT_TIMEOUT * 3)  # letting client set up and start writing
+
+    logger.info("Checking writes are running at all...")
+    assert cw.count_znodes(
+        parent=parent, hosts=hosts, username=helpers.USERNAME, password=password
+    )
+
+    logger.info("Removing leader pod...")
+    await helpers.delete_pod(ops_test=ops_test, unit_name=leader_name)
+    await asyncio.sleep(CLIENT_TIMEOUT)  # waiting for process to restore
+
+    logger.info("Checking leader IP changed...")
+    new_leader_host = helpers.get_unit_host(ops_test, leader_name)
+    assert new_leader_host != leader_host  # ensures pod was actually rescheduled
+
+    logger.info("Checking writes are increasing...")
+    writes = cw.count_znodes(
+        parent=parent, hosts=non_leader_hosts, username=helpers.USERNAME, password=password
+    )
+    await asyncio.sleep(CLIENT_TIMEOUT * 3)  # increasing writes
+    new_writes = cw.count_znodes(
+        parent=parent, hosts=non_leader_hosts, username=helpers.USERNAME, password=password
+    )
+    assert new_writes > writes, "writes not continuing to ZK"
+
+    logger.info("Checking leader re-election...")
+    new_leader_name = helpers.get_leader_name(ops_test, non_leader_hosts)
+    assert new_leader_name != leader_name
+
+    logger.info("Stopping continuous_writes...")
+    cw.stop_continuous_writes()
+
+    logger.info("Counting writes on surviving units...")
+    last_write = cw.get_last_znode(
+        parent=parent, hosts=non_leader_hosts, username=helpers.USERNAME, password=password
+    )
+    total_writes = cw.count_znodes(
+        parent=parent, hosts=non_leader_hosts, username=helpers.USERNAME, password=password
+    )
+    assert last_write == total_writes
+
+    logger.info("Checking old leader caught up...")
+    last_write_leader = cw.get_last_znode(
+        parent=parent, hosts=new_leader_host, username=helpers.USERNAME, password=password
+    )
+    total_writes_leader = cw.count_znodes(
+        parent=parent, hosts=new_leader_host, username=helpers.USERNAME, password=password
+    )
+    assert last_write == last_write_leader
+    assert total_writes == total_writes_leader
+
+
+@pytest.mark.abort_on_fail
+async def test_kill_db_process(ops_test: OpsTest, request):
```

Comment:
**question**: Isn't this one missing the pebble restart delay?

Reply:
We don't need it here I don't think, as we're hoping it actually restarts straight away anyway.

Reply:
All other _kill_db_process_ methods **do** have a restart delay, as the intention is to check that writes still continue when the leader went down and we have less remaining units.